packages/splits-sdk/src/types.ts

```diff
@@ -117,6 +118,174 @@ export type GetSplitBalanceConfig = {
   token?: string
 }
 
+// Warehouse
+export type WarehouseTransferConfig = {
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseTransferFromConfig = {
+  sender: Address
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseApproveConfig = {
+  spender: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseSetOperatorConfig = {
+  operator: Address
+  approved: boolean
+} & TransactionOverridesDict
+
+export type WarehouseInvalidateNonceConfig = {
+  nonce: bigint
+} & TransactionOverridesDict
+
+export type WarehouseTemporaryApproveAndCallConfig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+} & TransactionOverridesDict
+
+export type WarehouseTemporaryApproveAndCallBySigConfig = {
+  owner: Address
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+  nonce: bigint
+  deadline: number
+  signature: Hex
+} & TransactionOverridesDict
+
+export type WarehouseApproveBySig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  nonce: bigint
+  deadline: number
+}
+
+export type WarehouseTemporaryApproveAndCallBySig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+  nonce: bigint
+  deadline: number
+}
+
+export type WarehouseApproveBySigConfig = {
+  owner: Address
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  nonce: bigint
+  deadline: number
+  signature: Hex
+} & TransactionOverridesDict
+
+export type WarehouseDepositConfig = {
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseBatchDepositConfig = {
+  receivers: Address[]
+  token: Address
+  amounts: bigint[]
+} & TransactionOverridesDict
+
+export type WarehouseWithdrawConfig = {
+  owner: Address
+  token: Address
+} & TransactionOverridesDict
+
+export type WarehouseBatchWithdrawConfig = {
+  owner: Address
+  tokens: Address[]
+  amounts: bigint[]
+  withdrawer: Address
+} & TransactionOverridesDict
+
+export type WarehouseBatchTransferConfig = {
+  receivers: Address[]
+  token: Address
+  amounts: bigint[]
+} & TransactionOverridesDict
+
+export type WarehouseSetWithdrawConfig = {
+  incentive: number
+  paused: boolean
+} & TransactionOverridesDict
+
+export enum SplitV2Type {
+  Push,
+  Pull,
+}
+
+// Split V2
+export type CreateSplitV2Config = {
+  recipients: Address[]
+  allocations: bigint[]
+  distributionIncentive: number
+  splitType: SplitV2Type
+  owner: Address
+  creator?: Address
+  salt?: Hex
+} & TransactionOverridesDict
+
+export type UpdateSplitV2Config = {
+  split: Address
```

Comment:
nit: see you're using `split` as the argument name. I think I'm good with that. Given we're on a major version upgrade I think we can make a change to our formatting (up to now we've used `splitAddress`). Not sure if this change was intentional or not. Just wanted to highlight in case it wasn't.

Reply:
wasn't intentional, will update the PR to add splitAddress